src/Processor/Braintree/Hydrator/SubscriptionHydrator.php

```diff
@@ -93,4 +106,34 @@ private function hydratePlans(array $subscriptions): void
             $subscription->setPlan($hydrated);
         }
     }
+
+    /**
+     * Makes a request to Braintree and handles progressive back off in the case of network errors.
+     *
+     * @return mixed
+     */
+    private function braintreeRequestWithRetry(callable $fn, int $attempt = 0)
```

Comment:
https://github.com/braintree/braintree_node/issues/162#issuecomment-547111547

Braintree does not automatically retry failed transactions, and they intentionally leave it up to the merchant/customer to do their own retry logic because certain retries can lead to fraud flags being thrown in it's with a purchase.

So, I kept this just limited to the `SubscriptionHydrator` - but I can move this to a more generic place if you'd prefer then just import it here.

Reply:
This seems like it could be generally useful at a higher level, but maybe we just leave it here for now and keep it in our back pocket for a later date when it is needed elsewhere?